the-collab-lab/tcl-37-smart-shopping-list
# Add Header Navigation

## Description



This PR adds a nav element which persists at the top of the app. If the user clicks the "Home" link, they are redirected to the homepage, where they may sign into another list or return to their current one. If a token exists in state, a "Sign Out" button appears, which the user may use to exit their current list and create a new one - the existing token is removed from state and localStorage.

Also, refactored RouteLink component to remove the unnecessary Fragment wrapper.

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
| ✓  | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates

### Before

![header-before](https://user-images.githubusercontent.com/82783679/152441128-a34cb14f-3f55-4521-8502-b2e56c85cbcc.jpg)

### After

![header-preview](https://user-images.githubusercontent.com/82783679/152440625-5a7dbdf3-3730-4f0d-8d95-ebcb54a57f96.gif)


## Testing Steps / QA Criteria


1. Pull down branch `rb-header-nav`
2. `npm start`
3. If you have no token stored, create a new list.
4. Click on "HOME" link, try signing into an existing list, e.g. `vary adore suite`
5. Click on "Sign Out" button, app should return to fresh state
